### PR TITLE
Trait manager view changes

### PR DESCRIPTION
**Pull request by jlong-crestron**

### CODE REVIEW

File: src/filter/view/FilterView.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
+import { template } from 'underscore';
+import Backbone from 'backbone';
+
+const $ = Backbone.$;
+
+export default Backbone.View.extend({
+  bs: 'Backspace',
```
Comment (ntindle) [inline src/filter/view/FilterView.js:7]: Remove this

File: src/trait_manager/view/TraitsView.js [hunk 2]
```diff
@@ -41,6 +50,44 @@ export default DomainViews.extend({
...
     this.collection = comp ? comp.get('traits') : [];
+    this.collection.each &&
+      this.collection.each(function(model) {
+        model.set('visible', true);
+      });
+
     this.render();
+  },
+
+  searchCallBack(value) {
```
Comment (ntindle) [inline src/trait_manager/view/TraitsView.js:61]: Name this in a way that suggests we could implement other forms of search in the future (standardSearch, regexSearch, etc.)

File: src/block_manager/view/BlocksView.js [hunk 3]
```diff
@@ -63,6 +72,32 @@ export default Backbone.View.extend({
...
+   */
+  searchCallBack(value) {
+    var index = 1;
+    const processedValue = value.toLowerCase();
+    this.collection.models.forEach(model => {
+      const blockLabel = model.get('label').toLowerCase();
+      const categoryLabel = model.get('category').id;
+      if (
+        !blockLabel.includes(processedValue) &&
+        !categoryLabel.toLowerCase().includes(processedValue)
```
Comment (ntindle) [inline src/block_manager/view/BlocksView.js:86]: check if exists before to lowercase

**Review by ntindle — CHANGES_REQUESTED** [review]
Add more tests, fix issues

```console
_executeBound.js?3937:8 Uncaught TypeError: Cannot read property 'toLowerCase' of undefined
    at eval (webpack-internal:///./src/block_manager/view/BlocksView.js:100)
    at Array.forEach (<anonymous>)
    at child.searchCallBack (webpack-internal:///./src/block_manager/view/BlocksView.js:96)
    at child.handleChange (webpack-internal:///./src/filter/view/FilterView.js:43)
    at executeBound (_executeBound.js?3937:8)
    at HTMLDivElement.eval (bind.js?1bc3:10)
    at HTMLDivElement.eval (restArguments.js?f8bd:16)
    at HTMLDivElement.finalCallback (cash.esm.js?7230:767)
```
  | eval | @ | webpack-internal:///…w/BlocksView.js:100
-- | -- | -- | --
  | searchCallBack | @ | webpack-internal:///…ew/BlocksView.js:96
  | handleChange | @ | webpack-internal:///…ew/FilterView.js:43
  | executeBound | @ | _executeBound.js?3937:8
  | eval | @ | bind.js?1bc3:10
  | eval | @ | restArguments.js?f8bd:16
  | finalCallback | @ | cash.esm.js?7230:767

File: src/trait_manager/view/TraitsView.js [hunk 4]
```diff
@@ -41,6 +50,44 @@ export default DomainViews.extend({
...
+    var index = 1;
+    this.collection.forEach(element => {
```
[the remark this answers is not in the document]
Comment (jlong-crestron) [reply src/trait_manager/view/TraitsView.js:64]: Feature will not be included now, but in the future there should be a conversation about advanced search operations.

File: src/block_manager/view/BlocksView.js (hunk 3 above)
Comment (ntindle) [reply src/block_manager/view/BlocksView.js:86]: fix plz

File: src/filter/view/FilterView.js [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+    this.em = o.editor;
+    this.ppfx = ppfx;
+  },
+
+  template() {
+    return `
+      <div style="padding: .2em; display: flex; justify-content: flex-end;">
+        <div class=${this.holderClass()} style="justify-content: flex-start; width: 100%">
+        </div>
+        <span class="${this.ppfx}clm-tags-btn" data-clear>
```
[the remark this answers is not in the document]
Comment (ntindle) [reply src/filter/view/FilterView.js]: image
the x is not centered in the box at zoom

[the remark this answers is not in the document]
Comment (ntindle) [reply src/filter/view/FilterView.js]: I think this is fine in this case though

**Review by ntindle — APPROVED** [review]

File: src/filter/view/FilterView.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (jlong-crestron) [reply src/filter/view/FilterView.js]: Case where centering is off is not a desired case.

File: src/trait_manager/view/TraitsView.js (hunk 2 above)
Comment (jlong-crestron) [reply src/trait_manager/view/TraitsView.js]: basic search callback now has better naming.

**Review by ntindle — APPROVED** [review]

**Review by ntindle — APPROVED** [review]

File: src/filter/view/FilterView.js (hunk 1 above)
Comment (jlong-crestron) [reply src/filter/view/FilterView.js]: has been removed.

**Comment by ntindle** [comment]
LGTM

**Pull request merged**